### DISCUSSION TITLE
Multiple key directives

### DISCUSSION THREAD

**Discussion by DrAlan** [graphql-java/graphql-java#2224] [Q&A] [answered]
Can I use multiple @key directive?

I have schema like:
`type SomeType @key(fields: "field1") @key(fields: "field2") {`

When I try run application I get following exception:
`The directive 'key' MUST be defined as a repeatable directive if its repeated on an SDL element`

If I forcely define 'key' directive in my schema like that : `directive @key(fields: _FieldSet!) repeatable on OBJECT | INTERFACE` I get following:
`'key' is a repeatable directive and you have used a non repeatable access method`

Could you help me to coup with it?

**Comment by bbakerman** [comment, marked as answer]
You will only get this message if you call `graphql.DirectivesUtil#nonRepeatedDirectiveByNameWithAssert` - this method is in fact redundant and should have been been removed from graphql-java (it was an idea that was abandoned in PR but not cleaned up properly)

No production code references this method 

What version of graphql hava are you using?  What calls are you making?

`repeatable` directives is what you need (and its VERY new - the spec only just added it and we only just supported it)

**Reply by DrAlan** [reply]
Issue at federation-graphql-java-support lib. I use 0.6.1 and they have PR to fix that issue.
Thank you for your attention. Sorry.
